packages/core-manager/src/workers/actions/generate-log-zip.ts

```diff
@@ -0,0 +1,41 @@
+import archiver from "archiver";
+import { createWriteStream, ensureDirSync, renameSync } from "fs-extra";
+import { dirname, parse } from "path";
+import { Writable } from "stream";
+
+import { GenerateLog } from "./generate-log";
+import { IteratorToStream, LogTransformStream } from "./streams";
+
+export class GenerateLogZip extends GenerateLog {
+    public async execute(): Promise<void> {
+        const readStream = new IteratorToStream(this.database.getAllIterator("logs", this.options.query));
+        const writeStream = this.prepareOutputStream();
+
+        const archive = archiver("zip", {
+            zlib: { level: 9 },
+        });
+        archive.pipe(writeStream);
+
+        archive.append(
+            readStream.pipe(
+                new LogTransformStream({
+                    writableObjectMode: true,
```

Comment:
Why this option is present here, but not in `GenerateLogGz`?

Reply:
Zlib stream supports object mode, while archiver (zip) doesn't.